ServiceDescriptor.cs

```diff
@@ -45,18 +45,24 @@ public ServiceDescriptor()
         {
             // find co-located configuration xml. We search up to the ancestor directories to simplify debugging,
             // as well as trimming off ".vshost" suffix (which is used during debugging)
+            //Get the first parent to go into the recursive loop
             string p = ExecutablePath;
             string baseName = Path.GetFileNameWithoutExtension(p);
             if (baseName.EndsWith(".vshost")) baseName = baseName.Substring(0, baseName.Length - 7);
+            DirectoryInfo d = new DirectoryInfo(Path.GetDirectoryName(p));
             while (true)
             {
-                p = Path.GetDirectoryName(p);
-                if (File.Exists(Path.Combine(p, baseName + ".xml")))
+                if (File.Exists(Path.Combine(d.FullName, baseName + ".xml")))
                     break;
+
+                if (d.Parent == null)
+                    throw new FileNotFoundException("Unable to locate "+baseName+".xml file within executable directory or any parents");

```

Comment:
Sorry. My message was about the error message only, but I have not read the entire message 
"or any parents" was not visible in GitHub code viewer (and I have not noticed a scrollbar)